docs/api/field_type_type_and_value.md

```diff
@@ -0,0 +1,166 @@
+
+## Type class
+
+The Type class of a Field Type provides an implementation of the [`eZ\Publish\SPI\FieldType\FieldType`](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/SPI/FieldType/FieldType.php) interface.
+
+### Field Definition handling
+
+A custom Field Type is used in a Field definition of a custom Content Type.
+You can additionally provide [settings for the Field Type](#field-type-settings) and a [validator configuration](field_type_validation.md).
+Since the Public API cannot know anything about these, their handling is delegated to the Field Type itself through the following methods:
+
+#### `getFieldTypeIdentifier()`
+
+Returns a unique identifier for the custom Field Type which is used to assign the type to a Field definition.
+By convention it should be prefixed by a unique vendor shortcut (e.g. `ez` for eZ Systems).
+
+#### `getSettingsSchema()`
+
+This method retrieves via Public API a schema for the Field Type settings. A typical setting would be e.g. default value. The settings structure defined by this schema is stored in the `FieldDefinition`. Since it is not possible to define a generic format for such a schema, the Field Type is free to return any serializable data structure from this method.
+
+#### `getValidatorConfigurationSchema()`
+
+In addition to normal settings, the Field Type should provide schema settings for its validation process. The schema describes what kind of validation can be performed by the Field Type and which settings the user can specify to these validation methods. For example, the `ezstring` type can validate minimum and maximum length of the string. It therefore provides a schema to indicate to the user that they might specify the corresponding restrictions, when creating a `FieldDefinition` with this type. The schema does not underlie any regulations, except for that it must be serializable.
+
+#### `validateFieldSettings()`
+
+The type is asked to validate the settings (provided by the user) before the Public API stores those settings for the Field Type in a `FieldDefinition`. As a result, the Field Type must return if the given settings comply to the schema defined by `getSettingsSchema()`.
+
+#### `validateValidatorConfiguration()`
+
+As in `validateFieldSettings()`, this method verifies that the given validator configuration complies to the schema provided by `getValidatorConfigurationSchema()`.
+
+It is important to note that the schema definitions of the Field Type can be both of arbitrary and serializable format, it is highly recommended to use a simple hash structure.
+
+!!! note
+
+    Since it is not possible to enforce a schema format, the code using a specific Field Type must basically know all Field Types it deals with.
+
+This will also apply to all user interfaces and the REST API, which therefore must provide extension points to register handling code for custom Field Type. These extensions are not defined yet.
+
+### Field Type name
```

Comment:
I'm missing an explanation of what `eZ\Publish\SPI\FieldType\Nameable` is used for.

```
To be able to generate content name when field is part of name-schema or url-schema you have to implement `eZ\Publish\SPI\FieldType\Nameable` and register this service using the tag `ezpublish.fieldType.nameable`. 
```

Ref. https://github.com/ezsystems/ezpublish-kernel/blob/4eb8e38f020d97e7f8d8ff0e381db0e2b4141a1c/eZ/Publish/SPI/FieldType/Nameable.php